### PR TITLE
Auto-update nix archive

### LINKED ISSUE #296: Automatically update the nixpkgs archive we use
Once a week, use github actions to create a PR which just changes which version of nixpkgs unstable we use. The commit can be found at the bottom of this page: https://status.nixos.org/

### PR DESCRIPTION
**Pull request by devtomio**
closes #296

### CODE REVIEW

File: .github/workflows/update-nixpkgs-archive.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+          sed -i "/\/\/ Last Modified:/c\/\/ Last Modified: ${DATE}" src/nixpacks/plan/generator.rs
+          sed -i "/\/\/ https:\/\/github.com\/NixOS\/nixpkgs\/commit/c\/\/ $URL" src/nixpacks/plan/generator.rs
+          sed -i "/static NIXPKGS_ARCHIVE: &str/c\static NIXPKGS_ARCHIVE: &str = \"$SHA\";" src/nixpacks/plan/generator.rs
+
+          echo "::set-output name=sha::$SHA"
+          echo "::set-output name=url::$URL"
+
+      - name: Push Changes
+        uses: stefanzweifel/git-auto-commit-action@v4
```
Comment (Milo123459) [inline .github/workflows/update-nixpkgs-archive.yml:38]: This should make a pr

Comment (Milo123459) [reply .github/workflows/update-nixpkgs-archive.yml:38]: Just to ensure test passes. It just needs to make a pr like the update docker image one so we can review and merge

**Review by Milo123459 — APPROVED** [review]

File: .github/workflows/update-nixpkgs-archive.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: Update Nixpkgs Archive
+
+on:
+  schedule:
+    # At 00:00 on Sunday
+    - '0 0 * * 0'
```
Comment (coffee-cup) [inline .github/workflows/update-nixpkgs-archive.yml:6]: Can this be changed to `0 17 * * 1` (Monday at 5pm UTC)

**Review by coffee-cup — APPROVED** [review]
One small comment about changing the cron time but looks good! Thanks 😄

**Pull request merged**